android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java

```diff
@@ -279,16 +319,37 @@ private void initializePlayer() {
         }
     }
 
-    private MediaSource buildMediaSource(Uri uri, String overrideExtension) {
-        int type = Util.inferContentType(!TextUtils.isEmpty(overrideExtension) ? "." + overrideExtension
-                : uri.getLastPathSegment());
+    @SuppressWarnings("unchecked")
+    private MediaSource buildMediaSource(Uri uri, @Nullable String overrideExtension) {
+        @C.ContentType int type = Util.inferContentType(uri, overrideExtension);
         switch (type) {
             case C.TYPE_SS:
                 return new SsMediaSource(uri, buildDataSourceFactory(false),
                         new DefaultSsChunkSource.Factory(mediaDataSourceFactory), mainHandler, null);
             case C.TYPE_DASH:
-                return new DashMediaSource(uri, buildDataSourceFactory(false),
-                        new DefaultDashChunkSource.Factory(mediaDataSourceFactory), mainHandler, null);
+                return new DashMediaSource.Factory(
+                        new DefaultDashChunkSource.Factory(mediaDataSourceFactory),
+                        buildDataSourceFactory(false))
+                        .setManifestParser(
+                                new FilteringManifestParser<>(
+                                        new DashParser(new DashParser.Callback() {
+                                            @Override
+                                            public void onManifestParsed(com.google.android.exoplayer2.source.dash.manifest.DashManifest manifest) {
+                                                Log.d("Player", "onManifestParsed() manifest=" + manifest);
+
+                                                final ActionToken actionToken = ReactExoplayerView.this.actionToken;
+
+                                                if (manifest instanceof DashManifest && actionToken != null) {
+                                                    List kids = ((DashManifest) manifest).getDefaultKIds();
+                                                    Log.d("Player", "onManifestParsed() KIds=" + kids);
+                                                    String header = Utils.createXDrmInfoHeader(Utils.getSystem(actionToken.getDrmScheme()), kids);
+                                                    Log.d("Player", "onManifestParsed() XDrmHeader=" + header);
+                                                    drmCallback.setKeyRequestProperty("X-DRM-INFO", header);
+                                                }
+                                            }
+                                        }), (List<RepresentationKey>) getOfflineStreamKeys(uri)))
```

Comment:
Is the casting requried?

Reply:
No. I have changed the method signature so there is not casting now.